src/org/rascalmpl/values/parsetrees/TreeAdapter.java

```diff
@@ -266,6 +270,41 @@ else if (ProductionAdapter.isRegular(prod)) {
 						return null;
 				}
 			}
+			else if (ProductionAdapter.isError(prod)) {
+				int dot = ProductionAdapter.getErrorDot(prod);
+				IConstructor eprod = ProductionAdapter.getErrorProd(prod);
+				IList syms = ProductionAdapter.getSymbols(eprod);
+				int index = SymbolAdapter.indexOfLabel(syms, field);
+
+				if (index != -1) {
+					IConstructor sym = (IConstructor) syms.get(index);
+					sym = SymbolAdapter.stripLabelsAndConditions(sym);
+
+					if (dot <= index) {
+						// we have parsed the field so we can just return it.
+						// this is a likely scenario
+						return new FieldResult(sym, (ITree) tree.getArgs().get(index));
+					}
+					else {
+						// We have the right prodction and the field would be there, if we didn't recover from a parse error
+						// and are missing some of the children (including the indicated field).
+						// So we return a quasi tree that is of the right type, but it otherwise just an error tree
+						// like its parent. This tree does not have content, because we wouldn't know which of the skipped
+						// parts are meant.
+						var vf = IRascalValueFactory.getInstance();
+						// @PieterOlivier is this the right way to do this? I just need an empty tree of the right symbol.
+						var skipped = vf.constructor(RascalValueFactory.Production_Skipped);
+						var skippedProd = vf.constructor(RascalValueFactory.Production_Error, sym, skipped, vf.integer(0));
+						var skippedTree = vf.appl(skippedProd);
+						var parentLoc = getLocation(tree);
+						if (parentLoc != null) {
+							// TODO @PieterOlivier I guess we need the location of the last parsed element, and use an
+							// empty range that is just beyond that. But for now I'd like to test with this.
```

Comment:
You can use the location of the original `skipped` node (the last argument of `tree`) for that, it is guaranteed to have a location.

Reply:
ok great!

Reply:
I'm removing that code because it will just fail later and make things hard to debug.